db_master_deploy/deploy.sh

```diff
@@ -370,6 +371,52 @@ copy_table() {
     fi
 }
 
+#######################################
+# check locks
+# Globals:
+#   array_source
+# Arguments:
+#   None
+# Returns:
+#   None
+#######################################
+check_locks() {
+    # check if there are any lock files for the source objects
+    for source_object in "${array_source[@]}"; do
+        lockfile="${LOCK_DIR}/${source_object}.lock"
```

Comment:
yes the full name will be locked, so it will still be possible to deploy a single table while somebody else is deploying the whole database.